pkg/agents/team4/finalAgent/messages.go

```diff
@@ -0,0 +1,153 @@
+package team4EvoAgent
+
+import (
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/messages"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
+)
+
+func (a *CustomAgentEvo) SendingMessage() {
+
+	var msg messages.Message
+	floorToSend := a.Floor() + 1
+	switch a.params.messageCounter {
+	case 0:
+		msg = messages.NewAskFoodTakenMessage(a.ID(), a.Floor(), floorToSend)
+		a.SendMessage(msg)
+		a.params.sentMessages = append(a.params.sentMessages, msg)
+		a.Log("Team4 agent sent a message", infra.Fields{"message": msg.MessageType()})
```

Comment:
Ideally I'd move all of this to outside of the switch statement so you don't have to write the same thing this many times. But you'd need to handle the default case where `a.params.messageCounter > 9`

Reply:
we will change it in the next PR, if that's okay